src/ORM/DataObjectSchema.php

```diff
@@ -486,29 +500,65 @@ protected function cacheDatabaseFields($class)
     }
 
     /**
-     * Cache all indexes for the given class.
-     * Will do nothing if already cached
+     * Cache all indexes for the given class. Will do nothing if already cached.
      *
      * @param $class
      */
     protected function cacheDatabaseIndexes($class)
     {
-        if (array_key_exists($class, $this->databaseIndexes)) {
-            return;
+        if (!array_key_exists($class, $this->databaseIndexes)) {
+            $this->databaseIndexes[$class] = array_merge(
+                $this->defaultDatabaseIndexes($class),
+                $this->customDatabaseIndexes($class)
+            );
         }
+    }
+
+    /**
+     * Get "default" database indexable field types
+     *
+     * @param  string $class
+     * @return array
+     */
+    protected function defaultDatabaseIndexes($class)
```

Comment:
Should be called `cacheDefaultDatabaseIndexes` to distinguish this from the public getter and keep its purpose clear. This should also be the only place that defaultDatabaseIndexes is populated, since it seems part of this method has migrated to cacheDatabaseFields.